tests/benchmark/bm_vecsim_general.h

```diff
@@ -73,4 +88,66 @@ class BM_VecSimGeneral : public benchmark::Fixture {
     static inline std::string AttachRootPath(std::string file_name) {
         return std::string(getenv("ROOT")) + "/" + file_name;
     }
+
+    /** Mock tiered index callbacks. **/
+
+    static int submit_callback(void *job_queue, void *index_ctx, AsyncJob **jobs, JobCallback *CBs,
+                               size_t len) {
+        {
+            std::unique_lock<std::mutex> lock(queue_guard);
+            for (size_t i = 0; i < len; i++) {
+                static_cast<JobQueue *>(job_queue)->push(jobs[i]);
+            }
+        }
+        if (len == 1) {
+            queue_cond.notify_one();
+        } else {
+            queue_cond.notify_all();
+        }
+        return VecSim_OK;
+    }
+
+    // Main loop for background worker threads that execute the jobs form the job queue.
+    // run_thread uses as a signal to the thread that indicates whether it should keep running or
+    // stop and terminate the thread.
+    static void thread_main_loop() {
+        while (run_threads) {
+            std::unique_lock<std::mutex> lock(queue_guard);
+            // Wake up and acquire the lock (atomically) ONLY if the job queue is not empty at that
+            // point, or if the thread should not run anymore (and quit in that case).
+            queue_cond.wait(lock, []() { return !jobQ.empty() || !run_threads; });
+            if (!run_threads)
+                return;
+            auto job = jobQ.front();
+            jobQ.pop();
+            lock.unlock();
+            job->Execute(job);
+        }
+    }
+
+    static void thread_pool_join() {
+        // Check every 10 ms if queue is empty, and if so, terminate the threads loop.
+        while (true) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+            std::unique_lock<std::mutex> lock(queue_guard);
+            if (jobQ.empty()) {
+                run_threads = false;
+                queue_cond.notify_all();
+                break;
+            }
+        }
+        for (size_t i = 0; i < thread_pool_size; i++) {
+            thread_pool[i].join();
+        }
+        thread_pool.clear();
+    }
+
+    static void thread_pool_wait() {
```

Comment:
you can take the bitset from the binding's mock instead of re creating the threads